### ISSUE TITLE
[BOOM-3763] Ordering products in Catalog not working

### ISSUE THREAD

**Issue by prestonBot** [PrestaShop/PrestaShop#9903]
> This issue has been migrated from the Forge. Read the [original ticket here](http://forge.prestashop.com/browse/BOOM-3763).

- _**Reporter:**_ vincentbz
- _**Created at:**_ Fri, 1 Sep 2017 16:29:17 +0200

<p>Hi,</p>

<p>There are two (main) problems about ordering products in the catalog.</p>

<p>1. When I order them by drag & drop, click on <em>save & refresh</em> , they are saved in the opposite order. The ranking is reversed. For example if you don't change anything and save & refresh, the order is reversed.</p>

<p>2. When I save the order, the products in the category in Classic is not what i've defined, even if I have selected "Position inside category" in "default order  by" in the BO.</p>

<p>See images and I can show you if it's not clear.</p>

<p>Thanks</p>
        
* How to reproduce the issue ?

Go to catalog<br/>
Filter via a category, women for example (top right)<br/>
Click on &quot;reorder&quot;<br/>
Drag &amp; drop the products in the order you want<br/>
Take a screenshot of the order :p<br/>
Click on save and refresh<br/>
See, the order is the opposite<br/>

**Comment by prestonBot** [comment]
> This comment has been migrated from the Forge. Read the [original comment here](http://forge.prestashop.com/browse/BOOM-3763#comment-157627).

- _**Author:**_ bastien.bieri
- _**Created at:**_ Wed, 29 Nov 2017 11:44:28 +0100

<p>I'll have a look</p>

**Comment by prestonBot** [comment]
> This comment has been migrated from the Forge. Read the [original comment here](http://forge.prestashop.com/browse/BOOM-3763#comment-157646).

- _**Author:**_ bastien.bieri
- _**Created at:**_ Wed, 29 Nov 2017 15:45:24 +0100

<p><del>Looks like we have a fix candidate from contributor</del> <a href="http://forge.prestashop.com/secure/ViewProfile.jspa?name=sbordun" class="user-hover" rel="sbordun">Sergiy</a> <del>: <a href="https://github.com/PrestaShop/PrestaShop/pull/8458" class="external-link" rel="nofollow">https://github.com/PrestaShop/PrestaShop/pull/8458</a></del><br/>
 <del>Already reviewed and tested by me.</del><br/>
 <a href="http://forge.prestashop.com/secure/ViewProfile.jspa?name=vincentbz" class="user-hover" rel="vincentbz">Vincent Beudez</a> <del>can you please have a look on it ?</del></p>

<p>Edit : not good</p>

**Comment by prestonBot** [comment]
> This comment has been migrated from the Forge. Read the [original comment here](http://forge.prestashop.com/browse/BOOM-3763#comment-157725).

- _**Author:**_ vincentbz
- _**Created at:**_ Thu, 30 Nov 2017 15:48:12 +0100

<p>It's not related to the previous PR.</p>

<p><font color="#d04437"><b>Problems are :</b></font><br/>
 1. at first ordering, the first position is "0" and not "1". So, it fails. After the first ordering, it seems to be OK.<br/>
 2. when the option "display sub categories products" is activated in the Faceted search, the order on the FO doesn't work. It is correctly displayed only IF this option is not activated</p>

<p><font color="#14892c"><b>Solutions :</b></font><br/>
 1. The first position should be "1" and not "0"<br/>
 2. The order should work both when the option is activated or not</p>

<p>Thanks</p>

**Comment by prestonBot** [comment]
> This comment has been migrated from the Forge. Read the [original comment here](http://forge.prestashop.com/browse/BOOM-3763#comment-157987).

- _**Author:**_ cj22
- _**Created at:**_ Thu, 7 Dec 2017 08:55:28 +0100

<p>Thank you for the instructions on how to inverse everything back to the correct order!</p>

**Comment by prestonBot** [comment]
> This comment has been migrated from the Forge. Read the [original comment here](http://forge.prestashop.com/browse/BOOM-3763#comment-159750).

- _**Author:**_ marion_francois
- _**Created at:**_ Tue, 2 Jan 2018 14:55:30 +0100

<p>Small precision, sorting by quantity does not work if the module facetedsearch is activated.</p>

**Comment by prestonBot** [comment]
> This comment has been migrated from the Forge. Read the [original comment here](http://forge.prestashop.com/browse/BOOM-3763#comment-164718).

- _**Author:**_ fortuner
- _**Created at:**_ Thu, 22 Mar 2018 12:35:56 +0100

<p>Seriously? Fix on 1.7.4.X? You guys have to quit and go lawn makers</p>

**Comment by prestonBot** [comment]
> This comment has been migrated from the Forge. Read the [original comment here](http://forge.prestashop.com/browse/BOOM-3763#comment-164903).

- _**Author:**_ marion_francois
- _**Created at:**_ Mon, 26 Mar 2018 10:04:56 +0200

<p><a href="http://forge.prestashop.com/secure/ViewProfile.jspa?name=fortuner" class="user-hover" rel="fortuner">Igor Alexeyev</a></p>

<p>We are already a fairly small team (we're hiring!), and there's an insane amount of work before us. <br/>
But PrestaShop is an open source project, so it can be solved if you submit a pull request <img class="emoticon" src="http://forge.prestashop.com/images/icons/emoticons/smile.png" height="16" width="16" align="absmiddle" alt="" border="0"/></p>

**Comment by prestonBot** [comment]
> This comment has been migrated from the Forge. Read the [original comment here](http://forge.prestashop.com/browse/BOOM-3763#comment-164930).

- _**Author:**_ fortuner
- _**Created at:**_ Mon, 26 Mar 2018 12:33:59 +0200

<p>Sorry, Marion, but how many really useful pull requests were canceled because those were interfering with payed modules? Id's in urls is an example. And how many pulls are still pending to be checked? How many of them will be implemented in upcoming releases? 1.7.4? 1.7.5? Why not the 1.8 or 1.9?</p>

<p>I understand that developing CMS is a lot of work, but your superiors should think about attracting new Prestashop users, not selling modules to the existing ones. If there are happy and foremost successful Prestashop users, they will buy your modules by their own will, to make their business more successful. But not the opposite way.</p>

<p>And for this purpose Prestashop should have <b>at least basic functionality working</b></p>

<p>And for now we have a downgrade compared to 1.6:</p>
<ul>
	<li>no viewed products (the module still not fixed, besides there is working solution)</li>
	<li>no crosselling module (the same)</li>
	<li>no product comments, now we have to buy it</li>
	<li>attribute ids in product urls, that consequents to terrible SEO</li>
	<li>still no built-in blog!</li>
</ul>


<p>and so on</p>

<p><b>Believe me, I'm not offending you, Marion.</b></p>

<p>Just... Prestashop took the wrong way. </p>

<p>Here is an interesting article. It's not the best sign when your devotee have such thoughts.</p>

<p><a href="https://medium.com/@prestashoppoweruser/prestashop-a-sad-story-8b7e0a8795b8" class="external-link" rel="nofollow">https://medium.com/@prestashoppoweruser/prestashop-a-sad-story-8b7e0a8795b8</a></p>

**Comment by prestonBot** [comment]
> This comment has been migrated from the Forge. Read the [original comment here](http://forge.prestashop.com/browse/BOOM-3763#comment-165181).

- _**Author:**_ marion_francois
- _**Created at:**_ Wed, 28 Mar 2018 16:43:11 +0200

<p>Hello <a href="http://forge.prestashop.com/secure/ViewProfile.jspa?name=fortuner" class="user-hover" rel="fortuner">Igor Alexeyev</a></p>

<p>As I told you we are a small team and there's an insane amount of work before us.<br/>
 We want to improve the pull request process and simplify it in order to treat more of them.<br/>
 Our goal is to make each release significantly better than the one before.</p>

<p>The bugs are prioritized in our backlog and there are more important issues to solve before this one.<br/>
 Just so you know, there will be only 2 patches versions of 1.7.3 and we are already working on 1.7.4.0, so 1.7.4.x will arrive soon.<br/>
 You can see the release schedule here: <a href="http://build.prestashop.com/howtos/misc/2018-release-schedule/" class="external-link" rel="nofollow">http://build.prestashop.com/howtos/misc/2018-release-schedule/</a></p>

<p>Finally, your article date more than a year ago and lot of things have started changing in PrestaShop during this past 6-9 months:</p>
<ul>
	<li> We are taking a better way with our new CEO</li>
	<li>There's a new technical leadership that focuses on quality and reliability instead of time-to-market</li>
	<li>90% of the members of the core team were recruited in the last 6 months</li>
</ul>


<p>While this may or may not reassure you, we do the best we can with the team we have <img class="emoticon" src="http://forge.prestashop.com/images/icons/emoticons/smile.png" height="16" width="16" align="absmiddle" alt="" border="0"/></p>

**Comment by prestonBot** [comment]
> This comment has been migrated from the Forge. Read the [original comment here](http://forge.prestashop.com/browse/BOOM-3763#comment-166555).

- _**Author:**_ neich
- _**Created at:**_ Fri, 20 Apr 2018 20:03:22 +0200

<p>Any news about this issue? Solutions?</p>

<p>Regards</p>

**Comment by prestonBot** [comment]
> This comment has been migrated from the Forge. Read the [original comment here](http://forge.prestashop.com/browse/BOOM-3763#comment-166566).

- _**Author:**_ khouloud.belguith
- _**Created at:**_ Mon, 23 Apr 2018 09:58:33 +0200

<p>Hi neich,</p>

<p>Sorry, not yet.<br/>
It is added to our bug roadmap, it should be fixed in the PS 1.7.4.<br/>
But PrestaShop is an open source project, so it can be solved if you submit a pull request.</p>

<p>Best regards, Khouloud</p>

**Comment by anto7880** [comment]
With the version of prestashop 1.7.4.2 this problem still exists .... when it will be solved ??? :(

**Comment by khouloudbelguith** [comment]
Hi @anto7880,

I'm sorry but you are not alone and as I said, there are more important issues to solve before this one.
We are a fairly small team (we're hiring!), and there's an insane amount of work before us. But our goal is to make each release significantly better than the one before.
So, it will probably not solved before minimum a year. But PrestaShop is an open source project, so it can be solved before if someone submits a pull request to solve it.

Best regards, Khouloud

**Comment by lunaticatecnico** [comment]
I understand that you've had time enough to solve this issue. As Anto7880 said, the problem stills there un 1.7.4.2 and I think is kind of important to have the option to order products, it is a must in an ecommerce site. 

I think all is getting worse with 1.7 version, we were fans of Prestashop but considering all errors and bugs for 1.7 we are looking other options for our clients. 

Hope this issue gets a solution shortly, our clients are quite disappointed with this and other issues. 

Thanks

**Comment by marionf** [comment]
This issue seems to be fixed in 1.7.4.3 and with faceted search v2.2.0

![capture d ecran_432](https://user-images.githubusercontent.com/13449658/46747188-f1552680-ccb0-11e8-8fa9-e2025f53fd74.png)

![capture d ecran_433](https://user-images.githubusercontent.com/13449658/46747197-f4e8ad80-ccb0-11e8-82b4-cf14b3536d71.png)

![capture d ecran_434](https://user-images.githubusercontent.com/13449658/46747203-f87c3480-ccb0-11e8-9b5d-edeeffab582d.png)

**Comment by neich78** [comment]
I have Prestashop 1.7.4.3 and when filter in categories, the position column does not appear:
![captura de pantalla 2018-10-31 a las 10 49 53](https://user-images.githubusercontent.com/11721870/47780173-2c200c80-dcfb-11e8-80c5-785acba2d0b0.png)

**Comment by marionf** [comment]
@neich78 

I see it on your screenshot

![capture d ecran_545](https://user-images.githubusercontent.com/13449658/47792746-effda380-dd1c-11e8-985e-dab17aea22ce.png)

**Comment by neich78** [comment]
Yes, I see the position of each product but I cannot change it or drag it to move each item

**Comment by marionf** [comment]
You need to click on pedir de nuevo to change position

![capture d ecran_564](https://user-images.githubusercontent.com/13449658/48009000-18b9da80-e11b-11e8-971b-28e4e6d8552b.png)

**Comment by neich78** [comment]
OMG is true, sorry about that. Translation should be "Ordenar" and not "Pedir de nuevo". 
Thanks

**Comment by marionf** [comment]
@neich78 yes you are right, I have created the issue here: #11298

**Comment by gadnis** [comment]
> This issue seems to be fixed in 1.7.4.3 and with faceted search v2.2.0
> 
> ![capture d ecran_432](https://user-images.githubusercontent.com/13449658/46747188-f1552680-ccb0-11e8-8fa9-e2025f53fd74.png)
> 
> ![capture d ecran_433](https://user-images.githubusercontent.com/13449658/46747197-f4e8ad80-ccb0-11e8-82b4-cf14b3536d71.png)
> 
> ![capture d ecran_434](https://user-images.githubusercontent.com/13449658/46747203-f87c3480-ccb0-11e8-9b5d-edeeffab582d.png)

Hi, marionf,

Prestashop 1.7.4.4, default theme

This issue should not be marked as fixed, cause:
1. There's no title shown in FO (look at the pictures attached)
2. Default order by: Product quantity - is not working if ps_facedsearch module is enabled.

Please let me know if you're going to reopen it or I should create new issue ticket?

For more info please look at the pictures attached
![sorting-and-facedsearch-2](https://user-images.githubusercontent.com/10948341/49454183-52026a80-f7ed-11e8-9eb4-d0c17b79c142.jpg)
![sorting-and-facedsearch](https://user-images.githubusercontent.com/10948341/49454184-52026a80-f7ed-11e8-9cc3-a332a65f1862.jpg)
![screen shot 2018-12-04 at 17 44 29](https://user-images.githubusercontent.com/10948341/49454185-52026a80-f7ed-11e8-8ddb-e1b4863f21b1.png)

**Comment by marionf** [comment]
@gadnis 

Another issue already exists here: #9956

**Comment by gadnis** [comment]
> #9956

@marionf 

For me it looks like #9956 issue is about showing empty selectbox in category only. There's only some comments that sort by quantity is not working at all. Will developers understand that there's not only selectbox issue and will they try to solve order by quantity issue or should we open new issue about that?

By the way, sorting by modified date is working.

**Comment by marionf** [comment]
@gadnis 
It's not necessary to create a new issue 
In #9956 it's indicated that options don't work

**Comment by MariaRobles** [comment]
It still happening for me the same issue. 
I have prestashop 1.7.5
And búsqueda por facetas 3.4.0
Any one knows how to fix it???

**Comment by khouloudbelguith** [comment]
Hi @MariaRobles,

The problem is fixed if we use the icon in the position column. The product is selected and we can change the position and saved and all changes are validated.
Here's the PR: https://github.com/PrestaShop/PrestaShop/pull/14879 to fix the issue.
It is fixed in the PS1.7.7.0.
Since this PR correct the problem it's OK a New issue was created for the second issue: #15291

Thanks!

**Comment by MariaRobles** [comment]
Thanks a lot!

**Comment by eduardoserrano1987** [comment]
Hello all I have a new instalation of prestashop and have the same problem and can not find a way to solve it, i change position of product, but after in the website product dont move...

Can anybody help me!

Thanks

**Comment by loetje88** [comment]
Running on PrestaShop-versie: 1.7.6.4 , problem still persists.

Almost every button in back office no product page does not work by clicking, only after refresh.
Sorting backoffice sorting does not works FO.
